README.md

```diff
@@ -27,16 +27,16 @@ Elemenent types are in princple all Real types, but in practice this is limited
 * `PDMat`: full covariance matrix, defined as
 
 ```julia
-struct PDMat{T<:Real,S<:AbstractMatrix} <: AbstractPDMat{T}
+struct PDMat{T<:Real,S<:AbstractMatrix{T},F<:Factorization} <: AbstractPDMat{T}
     mat::S                      # input matrix
-    chol::Cholesky{T,S}         # Cholesky factorization of mat
+    fact::F                     # factorization of mat
```

Comment:
One needs an inner constructor that checks `isposdef(fact)` before calling `new`. Otherwise I could call
```julia
E = eigen(A)
PDMat{eltype(A),typeof(A),typeof(E)}(A, E)
```
even if `A` is not posdef.

Reply:
Currently none of the other constructors enforces positive semi-definite matrices: https://github.com/JuliaStats/PDMats.jl/issues/22

So ideally this should be addressed more generally, to ensure that such checks are performed by all `AbstractPDMat` types. But I wonder if it should be left for a separate PR?

Reply:
Sure. But it's not a hard change, we should just do it once other dust has settled.